src/solver/conv_hip_implicit_gemm_v4r4_gen_xdlops_wrw_fp32.cpp

```diff
@@ -0,0 +1,604 @@
+        std::string(" -DCK_PARAM_TUNABLE_GEMM_A_BLOCK_COPY_SRC_DATA_PER_READ_GEMM=") + std::to_string(GemmABlockCopySrcDataPerRead_GemmK) +
+        std::string(" -DCK_PARAM_TUNABLE_GEMM_B_BLOCK_COPY_DST_DATA_PER_WRITE_GEMM_N=") + std::to_string(GemmBBlockCopyDstDataPerWrite_GemmN) +
+        std::string(" -DCK_PARAM_TUNABLE_GEMM_A_BLOCK_COPY_DST_DATA_PER_WRITE_GEMM_M=") + std::to_string(GemmABlockCopyDstDataPerWrite_GemmM) +
+        std::string(" -DCK_USE_AMD_XDLOPS=") + (IsXdlopsSupport(ctx) ? '1' : '0') +
+        std::string(" -DCK_USE_AMD_XDLOPS_INLINE_ASM=") + (miopen::IsEnabled(MIOPEN_DEBUG_IMPLICIT_GEMM_XDLOPS_INLINE_ASM{}) ? '1' : '0') +
+        std::string(" -DCK_USE_AMD_XDLOPS_EMULATE=") + (miopen::IsEnabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_XDLOPS_EMULATE{}) ? '1' : '0') +
+        std::string(" -D__HIP_PLATFORM_HCC__=1") +
+        ctx.general_compile_options;
+    // clang-format on
+
+    result.construction_params.push_back(construction_parameters);
+    return result;
+}
+
+int ConvHipImplicitGemmV4R4GenXdlopsWrWFp32::RunAndMeasureSolution(miopen::Handle& profile_h,
+                                                                   ConstData_t bot_buf,
+                                                                   ConstData_t top_buf,
+                                                                   Data_t wei_buf,
+                                                                   ConstData_t bias_buf,
+                                                                   const ConvolutionContext& ctx,
+                                                                   const ConvSolution& solution,
+                                                                   float& elapsed_time) const
+{
+    assert(bias_buf == nullptr);
+    (void)bias_buf;
+
+    return RunAndMeasureSolutionBase(
+        profile_h, bot_buf, top_buf, wei_buf, ctx, solution, elapsed_time);
+}
+
+bool ConvHipImplicitGemmV4R4GenXdlopsWrWFp32::IsApplicable(const ConvolutionContext& ctx) const
+{
+    if(!(ctx.IsFp32()))
+        return false;
+
+    if(!ctx.direction.IsBackwardWrW())
+        return false;
+
+    if(!ctx.Is2d())
+        return false;
+
+    if(ctx.group_counts > 1)
+        return false;
+
+    const std::size_t n  = ConvolutionContextInterpreter::GetBatchN(ctx);
+    const std::size_t k  = ConvolutionContextInterpreter::GetOutputChannelK(ctx) / ctx.group_counts;
+    const std::size_t c  = ConvolutionContextInterpreter::GetInputChannelC(ctx) / ctx.group_counts;
+    const std::size_t y  = ConvolutionContextInterpreter::GetFilterHeightY(ctx);
+    const std::size_t x  = ConvolutionContextInterpreter::GetFilterWidthX(ctx);
+    const std::size_t ho = ConvolutionContextInterpreter::GetOutputHeightHo(ctx);
+    const std::size_t wo = ConvolutionContextInterpreter::GetOutputWidthWo(ctx);
+
+    const std::size_t GemmM = k;
+    const std::size_t GemmN = n * ho * wo;
```

Comment:
GemmM, GemmN and GemmK formulae are wrong

Reply:
Fixed. Thanks for finding it out.